hathor/p2p/sync_checkpoints.py

```diff
@@ -0,0 +1,213 @@
+# Copyright 2021 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from itertools import chain
+from typing import TYPE_CHECKING, Dict, List, Optional, Tuple
+
+from structlog import get_logger
+
+from hathor.checkpoint import Checkpoint
+from hathor.transaction.block import Block
+from hathor.transaction.storage.exceptions import TransactionDoesNotExist
+
+if TYPE_CHECKING:
+    from hathor.manager import HathorManager  # noqa: F401
+    from hathor.p2p.node_sync_v2 import NodeBlockSync  # noqa: F401
+
+logger = get_logger()
+
+
+class SyncCheckpoint:
```

Comment:
Missing docstring is almost all methods.

Reply:
Done here: https://github.com/HathorNetwork/hathor-core/pull/275/commits/a7fd898c9160f170a96feec65ba12c1bade819e9